internal/api/client/client.go

```diff
@@ -56,37 +56,8 @@ type Client struct {
 	Log *logger.Logger
 }
 
-// NewReadOnlyClient creates a new read-only client.
-// This client can be used to fetch objects but cannot write updates.
-func NewReadOnlyClient(ctx context.Context, region, bucket, distributionID string,
```

Comment:
I prefer the approach with an explicit constructor. IMO a ReadOnlyClient has a fundamental difference to a normal client, what is expressed by a separate constructor. I think such fundamental change in behavior shouldn't be made in a function parameter.

If you despite want to use boolean values as parameter, please define constant values in the same package so a caller can pass a meaningful value.

Reply:
If you just want to reduce code duplication (which would be a liitle overengineerin IMO), you could define a private function that is called by both constructors to hide the boolean parameter from callers.